app/src/main/java/com/xmartlabs/template/service/gson/RequiredFieldDeserializer.java

```diff
@@ -0,0 +1,50 @@
+package com.xmartlabs.template.service.gson;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Type;
+import java.util.Locale;
+
+import timber.log.Timber;
+
+/** {@link JsonDeserializer} that checks that all {@link GsonRequired} fields have a value */
+public class RequiredFieldDeserializer implements JsonDeserializer<Object> {
+  private Gson gson = new Gson();
+
+  @Override
+  public Object deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
+      throws JsonParseException {
+    Object instance = gson.fromJson(json, typeOfT);
+
+    for (Field field : instance.getClass().getDeclaredFields()) {
```

Comment:
The problem with using Stream/Exceptional here is that some exception i want to throw (and i want to do it as soon as a field fails), but the exceptions are captured inside the lambda abstraction.